steps/step1.py

```diff
@@ -65,55 +60,69 @@ def filter_by_rules(df) -> pd.DataFrame:
     dropped_months = 0
 
     # Set path for drop rules file
-    subdirectory_path = "parameters/drop_rules.csv"
-    drop_rules_path = os.path.join(parent_dir, subdirectory_path)
+    parent_directory = os.path.dirname(os.getcwd())
+    subdirectory_path = "GISTEMP_rewrite/parameters/drop_rules.csv"
+    drop_rules_path = os.path.join(parent_directory, subdirectory_path)
 
     # Read in drop rules csv, create copy of input dataframe
     df_drop_rules = pd.read_csv(drop_rules_path, skipinitialspace=True)
     df_filtered = df.copy()
-    
+
     # Read through each row in drop rules dataframe
     for _, row in df_drop_rules.iterrows():
-
         # Collect relevant information
-        station = row['Station_ID']
-        year_range = row['Omit_Period']
-        if '-' in year_range:
-            start_year = year_range.split('-')[0]
-            end_year = year_range.split('-')[1]
-            time_cols = [col for col in df_filtered.columns if not (col.startswith('Latitude') or col.startswith('Longitude'))]
-            if start_year == '0':
-
+        station = row["Station_ID"]
+        year_range = row["Omit_Period"]
+        if "-" in year_range:
+            start_year = year_range.split("-")[0]
+            end_year = year_range.split("-")[1]
+            time_cols = [
+                col
+                for col in df_filtered.columns
+                if not (col.startswith("Latitude") or col.startswith("Longitude"))
+            ]
+            if start_year == "0":
                 # Dropping all station data (ex: 0-9999)
-                if end_year == '9999':
+                if end_year == "9999":
                     df_filtered.loc[station, time_cols] = np.nan
                     dropped_months += len(time_cols)
 
                 # Dropping all values before a given year (ex: 0-1950)
                 else:
-                    cols_to_keep = [col for col in time_cols if int(col.split('_')[1]) > int(end_year)]
-                    cols_to_replace = [col for col in time_cols if col not in cols_to_keep]
-                    df_filtered.loc[station, cols_to_replace] = np.nan   
+                    cols_to_keep = [
+                        col
+                        for col in time_cols
+                        if int(col.split("_")[1]) > int(end_year)
+                    ]
+                    cols_to_replace = [
+                        col for col in time_cols if col not in cols_to_keep
+                    ]
+                    df_filtered.loc[station, cols_to_replace] = np.nan
```

Comment:
I have some thoughts on how to consolidate this code. But perhaps we can explore later on